algviz/interface/visitors.py

```diff
@@ -190,3 +190,29 @@ class WidgetVisitor(Visitor):
 
     def visit(self, *args, **kwargs):
         return super().visit(*args, **kwargs)
+
+class LinkedListVisitor(Visitor):
+    """Linked lists are represented as an array of an array and a current index, i.e.,
+    [[1, 2, 3], 0]."""
+    type_ = Tokens.ARRAY_T#treat this as an array, at least for now.
+
+    def has_current(self, ll):
+        return len(ll[0]) > ll[1] and ll[1] >= 0
+    
+    def has_next(self, ll):
+        return ((len(ll[0]) - 1) > ll[1]) and ll[1] >= 0
+
+    def get_item(self, ll):
```

Comment:
How about `get_data`?  The ArrayVisitor method was called `get_item` by an analogy with python's `__getitem__` that doesn't quite fit here.